avocado/test.py

```diff
@@ -598,6 +598,44 @@ def action(self):
         raise exceptions.TestNotFoundError(e_msg)
 
 
+class BuggyTest(Test):
+
+    """
+    Used when a test could not be imported (possibly an avocado buggy test, but it could be a buggy random py module).
+    """
+
+    def __init__(self, name=None, params=None, base_logdir=None, tag=None,
+                 job=None, runner_queue=None):
+        super(BuggyTest, self).__init__(name=name,
+                                        base_logdir=base_logdir,
+                                        params=params,
+                                        tag=tag, job=job,
+                                        runner_queue=runner_queue)
+
+    def action(self):
+        # pylint: disable=E0702
+        raise self.params.get('exception')
+
+
+class NotATest(Test):
+
+    """
+    Used when trying to run an importable module that has no Avocado Test classes inside of it.
+    """
+
+    def __init__(self, name=None, params=None, base_logdir=None, tag=None,
+                 job=None, runner_queue=None):
+        super(NotATest, self).__init__(name=name,
+                                       base_logdir=base_logdir,
+                                       tag=tag, job=job,
+                                       runner_queue=runner_queue)
+
+    def action(self):
+        e_msg = ('Python module %s does not contain an avocado test class. '
+                 'Please check your module for typos' % self.name)
```

Comment:
After thinking about this one, I'm going with

    'Python module %s does not contain an avocado test class ' % self.name